packages/search/graphql/resolvers/_queries/search.js

```diff
@@ -324,14 +325,29 @@ const parseOptions = (options) => {
 
 const MAX_NODES = 10000 // Limit, but exceedingly high
 
-const getFirst = (first, filter, user, recursive, unrestricted) => {
+const getFirst = (first, filter, user, recursive, forceUnrestricted) => {
   // we only restrict the nodes array
   // making totalCount always available
   // querying a single document by path is always allowed
   const path = getFilterValue(filter, 'path')
+
   const repoId = getFilterValue(filter, 'repoId')
   const oneRepoId = repoId && (!Array.isArray(repoId) || repoId.length === 1)
-  if (DOCUMENTS_RESTRICT_TO_ROLES && !recursive && !path && !oneRepoId && !unrestricted) {
+
+  const format = getFilterValue(filter, 'format')
+  const unrestricted =
+    format &&
+    format.length &&
+    DOCUMENTS_UNRESTRICTED_CHILDREN_REPO_IDS &&
```

Comment:
This seems to check that *some* unrestricted repo id is present in format filter. Should't it check that **every** repo id in in the format filter is present in `DOCUMENTS_UNRESTRICTED_CHILDREN_REPO_IDS`?

Reply:
no. format is augmented with repoIds of other environments.  
```
format: [
  'https://github.com/republik/format-preis-der-republik',
  'https://github.com/republik-test/format-preis-der-republik',
  'https://github.com/republik-dev/format-preis-der-republik',
  'https://github.com/republik-dev/format-preis-der-republik'
]
```

Reply:
Ah I see and multiple formats has it own key. So it's always one anyhow. Sorry didn't realize.

Reply:
no need to be sorry, it's absolutely not obvious.